scripts/build_automation/build-link.swift

```diff
@@ -0,0 +1,240 @@
+            "-H", "Content-Type: application/json; charset=utf-8",
+            "-H", "apikey: \(apiKey)",
+            "--data-binary", "@-"
+        ).input(withJSONObject: [
+            "title": title,
+            "destination": url,
+        ]).runJson(ShortenResponse.self)
+    }
+
+    static func getLinkDetails(id: String) throws -> ShortenResponse {
+        print(try cmd(
+            "curl", "-sf", "https://api.rebrandly.com/v1/links/\(id)",
+            "-H", "apikey: \(apiKey)"
+        ).runString())
+        return try cmd(
+            "curl", "-sf", "https://api.rebrandly.com/v1/links/\(id)",
+            "-H", "apikey: \(apiKey)"
+        ).runJson(ShortenResponse.self)
+    }
+}
+
+func getBuildLinkIDs(prID: String) throws -> [String]? {
+    let comments = try Github.getAllIssueComments(prID: prID)
+    let matches: [[String]] = try comments.compactMap { comment in
+        let body = comment.body
+        guard comment.user.login == "inst-danger" else { return nil }
+        let extractor = try NSRegularExpression(pattern: "<!-- \(magicString): (([^ -]+ )*)-->", options: [])
+        let range = NSRange(body.startIndex..<body.endIndex, in: body)
+        guard let match = extractor.firstMatch(in: body, options: [], range: range) else {
+            return nil
+        }
+        let linkRange = match.range(at: 1)
+        return body[Range(linkRange, in: body)!].split(separator: " ", omittingEmptySubsequences: true).map(String.init)
+    }
+    return matches.first
+}
+
+func createBuildLinks(prID: String) throws {
+    guard try getBuildLinkIDs(prID: prID) == nil else {
+        print("Links already exist, no work to be done")
+        return
+    }
+
+    let tempUrl = "https://github.com/\(repo)/pull/\(prID)"
+    var ids: [String] = []
+    var links: [String] = []
+    for app in App.allCases {
+        let result = try Rebrandly.shortenOrUpdate(url: tempUrl, title: app.title(forPr: prID))
+        ids.append(result.id)
+        links.append("\(app): [Link](https://\(result.shortUrl)) | [QR](\(result.qrUrl))")
+    }
+
+    let body = """
+        <!-- \(magicString): \(ids.joined(separator: " ")) -->
+        ## Latest Installable Builds
+
+        (will redirect to this page if no builds exist yet)
+        \(links.joined(separator: "\n"))
+        """
+    try Github.postComment(prID: prID, body: body)
+}
+
+func updateBuildLink(prID: String, app: App, url: String) throws {
+    guard let ids = try getBuildLinkIDs(prID: prID),
+        let appIndex = App.allCases.firstIndex(of: app),
+        ids.indices.contains(appIndex) else {
+        print("Couldn't find build link idendifier!")
+        return
+    }
+    try Rebrandly.shortenOrUpdate(id: ids[appIndex], url: url, title: app.title(forPr: prID))
+}
+
+let usage = """
+usage:
+    ./scripts/build_automation/build-link.swift generate-temp-links PR_NUMBER
+    ./scripts/build_automation/build-link.swift update-link student|teacher|parent BRANCH URL
+"""
```

Comment:
This script will be called by all 4 bitrise workflows. The tests workflow (triggered by PR) creates the comment with the links (using a URL shortener) if it doesn't already exist. Each of the app workflows then updates the destination of the short links to point to the latest build result.